query/PRoST-query/src/main/java/translator/Translator.java

```diff
@@ -361,21 +353,15 @@ private void removeTriplesFromGroups(final List<Triple> triples, final HashMap<S
 			entry.getValue().removeAll(triples);
 		}
 		// remove empty groups
-		final Iterator<Entry<String, List<Triple>>> it = groups.entrySet().iterator();
-		while (it.hasNext()) {
-			final HashMap.Entry<String, List<Triple>> pair = it.next();
-			if (pair.getValue().size() == 0) {
-				it.remove(); // avoids a ConcurrentModificationException
-			}
-		}
+        groups.entrySet().removeIf(pair -> pair.getValue().size() == 0);
```

Comment:
You can check is an entry (group) is empty directly after you remove triples (in the previous for loop) instead of having two loops.